pkg/api/customization/cluster/formatter.go

```diff
@@ -47,7 +47,19 @@ func (f *Formatter) Formatter(request *types.APIContext, resource *types.RawReso
 			resource.AddAction(request, v3.ClusterActionBackupEtcd)
 			resource.AddAction(request, v3.ClusterActionRestoreFromEtcdBackup)
 		}
-		resource.AddAction(request, v3.ClusterActionRunSecurityScan)
+		canCreateClusterScanObject := func() bool {
+			return request.AccessControl.CanDo(
+				v3.ClusterScanGroupVersionKind.Group,
+				v3.ClusterScanResource.Name,
+				"create",
+				request,
+				nil,
+				request.Schema,
+			) == nil
+		}
+		if canCreateClusterScanObject() {
+			resource.AddAction(request, v3.ClusterActionRunSecurityScan)
+		}
```

Comment:
I think we should add this after checking if the user can update the given cluster - so only cluster-owner of *this* cluster should have this action enabled when the cluster is viewed in API - and not any other cluster-owner(who possibly has the clusterId of other cluster)

Reply:
Good point. Will change.